bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/registry/AbstractRegistry.java

```diff
@@ -268,6 +279,15 @@ protected void addProvider(Provider<E> provider) {
         }
     }
 
+    /**
+     * Returns the key for a given element
+     *
+     * @param element
+     *            element
+     * @return key (must not be null)
+     */
+    protected abstract K getKey(E element);
```

Comment:
@kaikreuzer @sjka
WRT the comments above, I assume I found a "solution" for the element-key-topic that has very little influence about the existing API / code changes.
Please have a look at: maggu2810@9df148865fb46f3bf5cd6b399a1614a85705d706

If we really don't like an API breaking we could add a default implementation that returns the hashCode of the element, but as it is only about a lambda added to the constructor, I assume it is easy to adapt the proprietary registry classes.